mdata/notifierKafka/notifierKafka.go

```diff
@@ -82,6 +82,10 @@ func (c *NotifierKafka) start() {
 		if err != nil {
 			log.Fatalf("kafka-cluster: failed to get offset %d: %s", offsetTime, err)
 		}
+		if startOffset < 0 {
+			// happens when OffsetOldest or an offsetDuration was used and there is no message in the partition
```

Comment:
is this still accurate? AFAICT startOffset is the concrete (positive) offset, whereas offsetTime is now the "conceptual" (possibly negative) offset

Reply:
yes.   The comment clearly explains why this check is needed, ie `c.client.GetOffset()` will return '-1' if there are no messages in the partition.

Reply:
yeah the idea is that the concrete offset (startOffset) is not absolutely always positive as I imagined. It is negative in the specific case no message is in the queue.

Reply:
what if it was set to newest and there are no messages in the partition?

Reply:
In that case startOffset is set to 0